src/animated/Controller.ts

```diff
@@ -190,310 +198,322 @@ class Controller<DS extends object = any> {
     }
   }
 
-  // Execute the current queue of update props to our animations.
+  // Execute the current queue of prop updates.
   private _flush(onEnd?: OnEnd) {
     const { prevQueue } = this
-
-    // Updates can interrupt trailing queues, in that case we just merge values
-    if (prevQueue.length) {
-      const froms: any[] = []
-      const goals: any[] = []
-      prevQueue.forEach(({ from, to }) => {
-        if (is.obj(from)) froms.push(from)
-        if (is.obj(to)) goals.push(to)
-      })
-      prevQueue.length = 0
-
-      // Update `this.merged` with new values
-      if (froms.length) {
-        this.merged = Object.assign({}, ...froms, this.merged, ...goals)
-      } else if (goals.length) {
-        Object.assign(this.merged, ...goals)
-      }
-    }
-
-    // The guid helps when tracking frames, a new queue over an old one means an override.
-    // We discard async calls in that case
-    const guid = ++this.guid
     const queue = (this.prevQueue = this.queue)
     this.queue = prevQueue
+    prevQueue.length = 0
 
     // Track the number of running animations.
     let runsLeft = queue.length
-    this.runCount = runsLeft
+    this.runCount += runsLeft
```

Comment:
The `runCount` now accumulates across `_flush` calls. Since the `onRunEnd` callback will always be called, it wouldn't make sense to reset `runCount` on every flush.